packages/server/patches/http-proxy+1.18.1.patch

```diff
@@ -0,0 +1,61 @@
+diff --git a/node_modules/http-proxy/.auto-changelog b/node_modules/http-proxy/.auto-changelog
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/.gitattributes b/node_modules/http-proxy/.gitattributes
+deleted file mode 100644
+index 1a6bd45..0000000
+--- a/node_modules/http-proxy/.gitattributes
++++ /dev/null
+@@ -1 +0,0 @@
+-package-lock.json binary
+diff --git a/node_modules/http-proxy/CHANGELOG.md b/node_modules/http-proxy/CHANGELOG.md
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/CODE_OF_CONDUCT.md b/node_modules/http-proxy/CODE_OF_CONDUCT.md
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/LICENSE b/node_modules/http-proxy/LICENSE
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/README.md b/node_modules/http-proxy/README.md
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/codecov.yml b/node_modules/http-proxy/codecov.yml
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/index.js b/node_modules/http-proxy/index.js
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/lib/http-proxy.js b/node_modules/http-proxy/lib/http-proxy.js
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/lib/http-proxy/common.js b/node_modules/http-proxy/lib/http-proxy/common.js
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/lib/http-proxy/index.js b/node_modules/http-proxy/lib/http-proxy/index.js
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/lib/http-proxy/passes/web-incoming.js b/node_modules/http-proxy/lib/http-proxy/passes/web-incoming.js
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/lib/http-proxy/passes/web-outgoing.js b/node_modules/http-proxy/lib/http-proxy/passes/web-outgoing.js
+old mode 100644
+new mode 100755
+diff --git a/node_modules/http-proxy/lib/http-proxy/passes/ws-incoming.js b/node_modules/http-proxy/lib/http-proxy/passes/ws-incoming.js
+old mode 100644
+new mode 100755
+index 270f23f..feac117
+--- a/node_modules/http-proxy/lib/http-proxy/passes/ws-incoming.js
++++ b/node_modules/http-proxy/lib/http-proxy/passes/ws-incoming.js
+@@ -111,7 +111,7 @@ module.exports = {
+     proxyReq.on('error', onOutgoingError);
+     proxyReq.on('response', function (res) {
+       // if upgrade event isn't going to happen, close the socket
+-      if (!res.upgrade) {
++      if (!res.upgrade && socket.readyState === "open" && !socket.destroyed) {
```

Comment:
@AtofStryker Is this case sensitive? Would be a bit ridiculous if it is, but they are in all caps in the docs.

Reply:
which docs are you referring to? Are you saying the `readyState` is in caps?

Reply:
If this is a Websocket that conforms to MDN standards, `WebSocket.readyState` is a number that corresponds to `WebSocket.OPEN`, etc. It looks like socket-io also conforms to this. What's creating this socket, and what type is it? It doesn't seem to be defined in `http-proxy`, so it must be coming from our server code somewhere.

Reply:
However, `socket` is used as a duplex stream (I believe) in `checkMethodAndHeader`... what the heck is this value? 🤔

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/readyState

Reply:
Logging the value of `socket` here during the `websockets_spec` system test, and it looks like a decorated Duplex stream - it has stream related properties, like `destroyed`, `_readableState`, and `_writeableState`, but it also has properties describing a server:

```typescript
{
   ...restOfProperties,
   server: {
     ...somePrivateProps,
     allowHalfOpen: boolean
     connectionsCheckingInterval: number
     headersTimeout: number
     highWaterMark: number
     ...etc
   }
 }
 ```
 
 It does not, however, have a `readyState` property, so I'm not certain this conditional will ever evaluate as true.

Reply:
You need to log the value of `socket.readyState` directly. It's not enumerable and it returns `undefined` for `Object.getOwnPropertyDescriptor(socket, 'readyState')` but when accessed directly:
```
console.log('readyState is enumerable:', socket.propertyIsEnumerable('readyState'));
// => readyState is enumerable: false

console.log('description:', Object.getOwnPropertyDescriptor(socket, 'readyState'));
// => description: undefined

console.log('socket.readyState', typeof socket.readyState, socket.readyState);
// => socket.readyState string open
```

Reply:
It seems like this is the [Net.readyState](https://nodejs.org/api/net.html#socketreadystate) property